### PR TITLE
Reset internal state of avro parser

### LINKED ISSUE #38: Avro schema parser keep its internal state
The Avro schema parser has an internal state that it is not reseted in each call. Therefore, if compendium receives two protocols with the same name in different HTTP calls it will return the following error: 
```
org.apache.avro.SchemaParseException: Can't redefine
```

### PR DESCRIPTION
**Pull request by pirita**
Fixes #38

### CODE REVIEW

File: modules/server/src/test/scala/higherkindness/compendium/core/ProtocolUtilsSpec.scala [hunk 1]
```diff
@@ -44,6 +44,16 @@ object ProtocolUtilsSpec extends Specification with ScalaCheck {
         .validateProtocol(protocol)
         .unsafeRunSync must throwA[org.apache.avro.SchemaParseException]
     }
+
+    "It is possible to parse multiple protocols" >> {
+      val stream: InputStream = getClass.getResourceAsStream("/correct.avro")
+      val text: String        = scala.io.Source.fromInputStream(stream).getLines.mkString
+      val protocol: Protocol  = Protocol(text)
+
+      utils.validateProtocol(protocol).unsafeRunSync
```
Comment (juanpedromoreno) [inline modules/server/src/test/scala/higherkindness/compendium/core/ProtocolUtilsSpec.scala]: Is this needed?

Comment (pirita) [reply modules/server/src/test/scala/higherkindness/compendium/core/ProtocolUtilsSpec.scala]: The problem occurs when making a second call with a protocol with a similar name for a record. We can refactor the code in the following way to make only one `unsafeRunSync `

```scala

      utils
        .validateProtocol(protocol)
        .flatTap(_ => utils.validateProtocol(protocol))
        .unsafeRunSync should not(throwA[org.apache.avro.SchemaParseException])
```

Comment (pepegar) [reply modules/server/src/test/scala/higherkindness/compendium/core/ProtocolUtilsSpec.scala]: +1 to that change. You can even:

`(utils.validateProtool(protocol) *> utils.validateProtocol(protocol)).unsafeRunSync`

**Comment by pepegar** [comment]
👍 merge on green :)

**Pull request merged**